cdap-proto/src/main/java/co/cask/cdap/proto/BasicThrowable.java

```diff
@@ -85,7 +87,41 @@ public String getMessage() {
    * Return the cause of Throwable if it is available, otherwise {@code null}.
    */
   @Nullable
-  public WorkflowNodeThrowable getCause() {
+  public BasicThrowable getCause() {
     return cause;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    BasicThrowable that = (BasicThrowable) o;
+
+    if (!className.equals(that.className)) {
+      return false;
+    }
+    if (!message.equals(that.message)) {
+      return false;
+    }
+
+    if (!Arrays.equals(stackTraces, that.stackTraces)) {
+      return false;
+    }
+    return !(cause != null ? !cause.equals(that.cause) : that.cause != null);
```

Comment:
you're right; that does read more easily.
This is just what intellij auto-generated. Updated.